src/main/java/com/blackhearth/blockchain/SimpleBlockchainApplication.java

```diff
@@ -10,8 +13,14 @@
 @SpringBootApplication
 public class SimpleBlockchainApplication {
 
+	@SneakyThrows
 	public static void main(String[] args) {
 		SpringApplication.run(SimpleBlockchainApplication.class, args);
 		Map<RunParams, Object> params = ParamsReader.getParams(args);
+
+		BasicPeerToPeerService basicPeerToPeerService = new BasicPeerToPeerService(new InterpreterBlockChainCommunication());
```

Comment:
To jest bardzo dziwna konstrukcja w spring boocie.

Istnieje klasa BlockChainNode która wywołuje p2pService.start().

Zamiast tej dziwnej konstrukcji:
1. Stwórz beana BasicPeerToPeerService za pomocą @Component.
2. W public static void main(String[] args) pobierz context aplikacji (zwraca go funkcja SpringApplication.run(...);
3. Użyj contextu aby zdobyć beana BlockChainNode. (metoda getBean)
4. Użyj funkcji start na tym singletonie.
5. upewnij się, że MessageFactory z pakietu protocol dostanie BlockChainNodeData (ale z innej funkcji niż start). Również BlockChainNode powinien ogłosić się sieci (wysłać odpowiedni komunikat za pomocą protokołu (patrz dokumentacja).